### DISCUSSION TITLE
Coverage: ИНН плательщика и другие атрибуты

### DISCUSSION THREAD

**Discussion by eugenekogan-spb** [fhir-ru/core#152] [Административная РГ]
По сообщению А.Звягина на встрече http://fhir-ru.zendoc.me/meetings.2022-05-31, при моделировании лабораторного CDA в части информации о плательщике возникли следующие варианта хранения ИНН:
Вариант 1. В ресурсе Patient/Person, Organisation в зависимости от типа плательщика
Вариант 2. В самом Coverage
Вариант 3. И там, и там.

Аргументы в пользу варианта 2 не представлены, кроме того, что ИНН плательщика является обязательным полем договора, и договор моделируется ресурсом Coverage.

Вариант 3 имеет существенный минус - информация дублируется и есть риск расхождения информации.

При использовании варианта 1 необходимо обеспечить обязательность заполнения ИНН в Patient/Person, Organisation при их использовании в качестве плательщиков.

Предлагается: 
- сделать специальный "финансовый" профиль для ресурсов Patient/Person, Organisation, профилировать только атрибут Identifier, указать, что он имеет кардинальность 1..* 
- в профиле Coverage в атрибуте Payer указать именно эти профили.

**Comment by eugenekogan-spb** [comment]
Другие замечания о Coverage
(из сообщения @uestas в чате ОДЛИ 06/06/2020)

Перечень комментариев и предложений по структуре, представленной во вкладке “Coverage” файла “ПЛИ CDA 20220530.xslx”
1.	В спецификации не указан элемент "beneficiary", возможно стоит его включить.
2.	В спецификации не указан элемент "period", возможно стоит его включить.
3.	Тип идентификатора не подходит (NIIP), это идентификатор страховой, а нам нужен идентификатор полиса, описание:
 <img width="445" alt="image" src="https://user-images.githubusercontent.com/62525329/172210645-57f0b07c-ae1f-4e9e-b7ad-5032bcdec502.png">
4.	Исходя из вышесказанного, расширяя справочник, можно добавить и серию полиса, как значение справочника типов, тогда мы покроем все известные на текущий момент случаи.

**Comment by eugenekogan-spb** [comment]
Из решений на созвоне административной группы: http://fhir-ru.zendoc.me/meetings.2022-06-03-adm
1. В примере Coverage.Identifier кроме Type указать System, Display, иначе непонятно откуда код взят и что он означает
2. В профиле Coverage – оставить предложенную А.Звягиным нотацию типов и подтипов документов с помощью атрибута Class. Тип документа = (Полис ОМС, Полис ДМС, платные). Подтип – вводится только для полиса ОМС.
3. В примере Coverage для class.type.code = ‘class’ указать display = ‘Вид документа’, class.type.value = ‘Полис ОМС’.
4. В примере Coverage для class.type.code = ‘subclass’ указать display = ‘Вид полиса ОМС’, class.type.value = ‘Полис ОМС единого образца, бессрочный’.

**Comment by eugenekogan-spb** [comment]
На созвоне административной группы от 15.07 было высказано возражение к пп.2,3,4 предыдущего решения, и альтернативное предложение. Кодирование в Coverage.Class выглядит не соответствующим смыслу, поскольку никто не классифицирует страховые покрытия по виду представленного документа.
Мы разделяем страховые покрытия по источнику финансирования – за счет ОМС, ДМС, средств физических и юридических лиц. Эти варианты кодируются атрибутом Coverage.Type. Каждый вариант имеет особенности, которые могут быть установлены с помощью slicing по полю type.
Сведения о представленном документе (полисе), подтверждающем наличие покрытия, более естественно отразить атрибутом Coverage. Contract. В профиле следует закодировать, что если Coverage.Type = Средства ОМС, то Contract.Type выбирается из справочника видов полисов ОМС и т.д.

**Reply by Vrodvi** [reply]
> Кодирование в Coverage.Class выглядит не соответствующим смыслу

Согласен, этот атрибут лучше не использовать.

> Мы разделяем страховые покрытия по источнику финансирования – за счет ОМС, ДМС, средств физических и юридических лиц. Эти варианты кодируются атрибутом Coverage.Type. Каждый вариант имеет особенности, которые могут быть установлены с помощью slicing по полю type.

Поддерживаю. Только в этом месте справочник определён как Preffered и значения в нём адекватные, так что, видимо, нужно будет маппирование?

> Сведения о представленном документе (полисе), подтверждающем наличие покрытия, более естественно отразить атрибутом Coverage. Contract. В профиле следует закодировать, что если Coverage.Type = Средства ОМС, то Contract.Type выбирается из справочника видов полисов ОМС и т.д.

Выглядит логично. Интересно посмотреть на формализованное описание взаимосвязи атрибутов в двух взаимосвязанных ресурсах (Coverage.Type и Contract.Type) (видимо аналогичные ограничения нужны ещё и для атрибута `payor` - в зависимости от типа оплаты это должна быть либо страховая организация, либо юр.лицо, либо физ. лицо с соответствующими профилями). 
Также подобный подход позволит для случая оплаты средствами физических лиц указать в Contract информацию о договоре на оказание платных медицинских услуг.

Также хотелось бы группа подготовила примеры ресурсов для всех обозначенных вариантов оплаты - за счет ОМС, ДМС, средств физических и юридических лиц - для большей наглядности и чтобы обсуждать было проще.
